components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -565,12 +571,15 @@ during post consent handling to get the original ConsentClaimsData object (Assum
                 getSSOConsentService().processConsent(approvedClaimIds, serviceProvider,
                         loggedInUser, value, false);
             }
-        } catch (OAuthSystemException | SSOConsentServiceException e) {
+
+        } catch (OAuthSystemException | SSOConsentServiceException | RequestObjectException |
+                ClaimMetadataException e) {
```

Comment:
Shall we seperatly catch the two exceptions and add specific error messages.

Reply:
Added.